MetadataExtractor/IO/BufferScope.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Drew Noakes and contributors. All Rights Reserved. Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+using System.Buffers;
+
+namespace MetadataExtractor.IO;
+
+internal ref struct BufferScope
```

Comment:
Could add implicit conversion to a `Span<byte>` with:

```c#
    public static implicit operator Span<byte>(BufferScope bufferScope)
    {
        return bufferScope._span;
    }
```

That could clean up the usages a little. I'd also consider naming this `ScopedBuffer` as I think it's more of a buffer than a scope.